File: examples/infolder/backend/go/models/tag.go
package models

type Tag struct {
	Id string `gorm:"primaryKey;column:id" json:"id"`
	Name string `gorm:"column:name;not null;uniqueIndex" json:"name"`
	Slug string `gorm:"column:slug;not null;uniqueIndex" json:"slug"`
	Posts []PostTag `gorm:"foreignKey:TagId" json:"posts"`
}

File: examples/outfolder/example/lib/models/tag/tag.dart
import 'package:schemix/schemix.dart';

import '../post_tag/post_tag.dart';

part 'tag.schemix.dart';

@Schemix(
  name: 'tags',
  schemaVersion: 1,
  enableTimestamps: false,
  enableSoftDelete: false,
)
class Tag {
  @PrimaryKey(autoGenerate: true)
  final String id;

  @Unique()
  @Length(min: 1, max: 50)
  final String name;

  @SchemixField(description: 'URL-friendly identifier for the tag')
  @Unique()
  final String slug;

  @HasMany(PostTag)
  final List<PostTag> posts;

  const Tag({required this.id, required this.name, required this.slug, this.posts = const []});

  factory Tag.fromJson(Map<String, dynamic> json) => _$TagFromJson(json);

  Map<String, dynamic> toJson() => _$TagToJson(this);

  Tag copy() => _$TagCopy(this);
}
